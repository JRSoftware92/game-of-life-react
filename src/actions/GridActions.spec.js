import {
    randomizeGrid,
    clearGrid,
    toggleTile,
    startRunningLife,
    stopRunningLife,
    runLifeIteration,
    selectRule,
} from './GridActions';

describe('Grid Action Creator Functions', () => {
   describe('Action Shapes', () => {
      test('Randomize Grid Action', () => {
          expect(randomizeGrid(2, 2, 50)).toMatchSnapshot();
      });

       test('Clear Grid Action', () => {
           expect(clearGrid(2, 2)).toMatchSnapshot();
       });

       test('Toggle Tile Action', () => {
           expect(toggleTile(2, 2)).toMatchSnapshot();
       });

       test('Start Life Simulation Action', () => {
           expect(startRunningLife()).toMatchSnapshot();
       });

       test('Stop Life Simulation Action', () => {
           expect(stopRunningLife()).toMatchSnapshot();
       });

       test('Run Single Life Simlation Iteration Action', () => {
           expect(runLifeIteration()).toMatchSnapshot();
       });

       test('Select Simulation Rule Action', () => {
           expect(selectRule({ value: 'selectedRule' })).toMatchSnapshot();
       });
   });
});
