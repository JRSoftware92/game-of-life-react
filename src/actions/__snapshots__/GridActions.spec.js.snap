// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Action Creator Functions Action Shapes Clear Grid Action 1`] = `
Object {
  "height": 2,
  "type": "CLEAR",
  "width": 2,
}
`;

exports[`Grid Action Creator Functions Action Shapes Randomize Grid Action 1`] = `
Object {
  "density": 50,
  "height": 2,
  "type": "RANDOM",
  "width": 2,
}
`;

exports[`Grid Action Creator Functions Action Shapes Run Single Life Simulation Iteration Action 1`] = `
Object {
  "type": "NEXT",
}
`;

exports[`Grid Action Creator Functions Action Shapes Select Simulation Rule Action 1`] = `
Object {
  "selectedRule": Object {
    "value": "selectedRule",
  },
  "type": "SELECT_RULE",
}
`;

exports[`Grid Action Creator Functions Action Shapes Start Life Simulation Action 1`] = `
Object {
  "type": "START",
}
`;

exports[`Grid Action Creator Functions Action Shapes Stop Life Simulation Action 1`] = `
Object {
  "type": "STOP",
}
`;

exports[`Grid Action Creator Functions Action Shapes Toggle Tile Action 1`] = `
Object {
  "type": "TOGGLE_TILE",
  "x": 2,
  "y": 2,
}
`;
